asdf_astropy/tags/transform/polynomial.py

```diff
@@ -0,0 +1,301 @@
+                        coeffs[name] = coefficients[i, j]
+            model = models.Polynomial2D(degree,
+                                        x_domain=x_domain,
+                                        y_domain=y_domain,
+                                        x_window=x_window,
+                                        y_window=y_window,
+                                        **coeffs)
+        else:
+            raise NotImplementedError(
+                "Asdf currently only supports 1D or 2D polynomial transform.")
+        return model
+
+    def to_yaml_tree(self, model):
+        if isinstance(model, models.Polynomial1D):
+            coefficients = np.array(model.parameters)
+        elif isinstance(model, models.Polynomial2D):
+            degree = model.degree
+            coefficients = np.zeros((degree + 1, degree + 1))
+            for i in range(degree + 1):
+                for j in range(degree + 1):
+                    if i + j < degree + 1:
+                        name = 'c' + str(i) + '_' + str(j)
+                        coefficients[i, j] = getattr(model, name).value
+        node = {'coefficients': coefficients}
+        ndim = model.n_inputs
+
+        if self.version >= PolynomialTypeBase.DOMAIN_WINDOW_MIN_VERSION:
+            # Schema versions prior to 1.2 included an unrelated "domain"
+            # property.  We can't serialize the new domain values with those
+            # versions because they don't validate.
+            if ndim == 1:
+                if model.domain is not None:
+                    node['domain'] = model.domain
+                if model.window is not None:
+                    node['window'] = model.window
+            else:
+                if model.x_domain or model.y_domain is not None:
+                    node['domain'] = (model.x_domain, model.y_domain)
+                if model.x_window or model.y_window is not None:
+                    node['window'] = (model.x_window, model.y_window)
+
+        return node
+
+    def assert_equal(self, a, b):
+        # TODO: If models become comparable themselves, remove this.
+        super().assert_equal(a, b)
+        assert (isinstance(a, (models.Polynomial1D, models.Polynomial2D)) and
+                isinstance(b, (models.Polynomial1D, models.Polynomial2D)))
+        assert_array_equal(a.parameters, b.parameters)
+
+        if cls.version > PolynomialTypeBase.DOMAIN_WINDOW_MIN_VERSION:
+            # Schema versions prior to 1.2 are known not to serialize
+            # domain or window.
+            if isinstance(a, models.Polynomial1D):
+                assert a.domain == b.domain
+                assert a.window == b.window
+            else:
+                assert a.x_domain == b.x_domain
+                assert a.x_window == b.x_window
+                assert a.y_domain == b.y_domain
+                assert a.y_window == b.y_window
+
+
+class PolynomialType1_0(PolynomialTypeBase):
```

Comment:
Ha!  I forgot about these.  I did this because I didn't understand how supported_versions worked in the old framework.  We can merge these back into one class again and just list multiple `tags`.